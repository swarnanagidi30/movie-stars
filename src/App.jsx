import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import './App.scss';
import CountrySelect from './components/CharacterSelect';
import TextField from '@material-ui/core/TextField';
import MoviesList from './components/MoviesList';


const useStyles = makeStyles((theme) => ({
  appItem: {
    marginTop: theme.spacing(3),
    width: '25%'
  },
  textBox: {
    width: '100%'
  }
}));

function App() {

  const classes = useStyles();
  const [character, setCharacter] = React.useState({});
  const [latestMovie, setLatestMovie] = React.useState(null);

  const handleChange = (value) => {
    setCharacter(value);
  };
  const handleLatestMovie = (value) => {
    setLatestMovie(value);
  }

  const getYear = (dateString) => {
    const date = new Date(dateString);
    return date.getFullYear();
  };

  return (
    <div className="App">
      <header className="App-header">
        <div className={classes.appItem}>
          <CountrySelect onChange={handleChange} />
        </div>
        <div className={classes.appItem}>
          <MoviesList films={character && character.films} onLatestMovie={handleLatestMovie} />
        </div>
        <div className={classes.appItem}>
          <TextField className={classes.textBox} label="Name / Year last Movie:" value={latestMovie ? (latestMovie.title + ' - ' + getYear(latestMovie.release_date)) : ''} />
        </div>
      </header>
    </div>
  );
}

export default App;
