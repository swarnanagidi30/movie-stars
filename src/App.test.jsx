import React from 'react';
jest.mock('node-fetch');

// import fetch, { Response } from 'node-fetch';
// import { render } from '@testing-library/react';

import { createShallow } from '@material-ui/core/test-utils';
import { ThemeProvider } from '@material-ui/core/styles';
import App from './App';
import toJson from "enzyme-to-json"

// test('renders Name / Year last Movie link', () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/Name \/ Year last Movie/i);
//   expect(linkElement).toBeInTheDocument();
// });




describe("<App />", () => {
  const renderShallowUntilComponent = createShallow({ "untilSelector": "App" })
  let Component
  beforeEach(() => {

    Component = renderShallowUntilComponent(<ThemeProvider theme={{ success: { main: '#fff' } }}><App /></ThemeProvider>)
  })

  it("loads", () => {
    expect(typeof Component).toBe("object")
  })

  it("has a <CharacterSelect> container", () => {
    expect(Component.find("CharacterSelect").length).toBeGreaterThanOrEqual(1)
  })

  it("has a <MoviesList> container", () => {
    expect(Component.find("MoviesList").length).toBeGreaterThanOrEqual(1)
  })

  it("has a <TextField> container", () => {
    expect(Component.find("WithStyles(ForwardRef(TextField))").length).toBeGreaterThanOrEqual(1)
  })

  it("snapshots", () => {
    expect(toJson(Component)).toMatchSnapshot()
  })
})
