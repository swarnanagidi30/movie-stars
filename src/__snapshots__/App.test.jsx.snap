// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshots 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="makeStyles-appItem-1"
    >
      <CharacterSelect
        onChange={[Function]}
      />
    </div>
    <div
      className="makeStyles-appItem-1"
    >
      <MoviesList
        onLatestMovie={[Function]}
      />
    </div>
    <div
      className="makeStyles-appItem-1"
    >
      <WithStyles(ForwardRef(TextField))
        className="makeStyles-textBox-2"
        label="Name / Year last Movie:"
        value=""
      />
    </div>
  </header>
</div>
`;
